pkg/cli/login/loginoptions.go

```diff
@@ -266,6 +267,27 @@ func (o *LoginOptions) gatherProjectInfo() error {
 		return err
 	}
 
+	canRequest, err := loginutil.CanRequestProjects(o.Config, o.DefaultNamespace)
+	if err != nil {
+		return err
+	}
+	// check if ProjectRequestMessage is set
+	if !canRequest {
+		msg := ""
+		res, err := projectClient.RESTClient().Get().Resource("projectrequests").DoRaw(context.TODO())
```

Comment:
```
status := metav1.Status{}
projectClient.RESTClient().Get().Resource("projectrequests").Do(context.TODO()).Into(&status)
```
will drop the entire encoding code from here.

Reply:
The issue with that is that here:
https://github.com/openshift/oc/blob/master/vendor/k8s.io/client-go/rest/request.go#L920-#L936
The resp.body is transformed into the error, so you lose the status with the projectRequestMessage.  Then, if you instead try to use DoRaw you'll get the response body, but as []byte which cannot be used in `(r Result) Into(obj runtime.Object)` :( I tried, but the `type Result` has only unexported fields.